### PR TITLE
Prepare v1.7.0-rc2

### PR DESCRIPTION
**Pull request by jrick**
Updates dcrd to release-v1.7.0-rc2 tag.

Updates dcrwallet to release-v1.7.0-rc2 tag.

Updates dcrdex and dcrdex-assets to to v0.4.0-rc2 tags.

Updates dcrlnd to master branch.

### CODE REVIEW

**Review by alexlyp — APPROVED** [review]

File: main.go [hunk 1]
```diff
@@ -148,8 +148,8 @@ var dists = []dist{{
 	ldflags: fmt.Sprintf(`-buildid= -s -w `+
 		`-X decred.org/dcrdex/client/cmd/dexc/version.appBuild=release `+
 		`-X decred.org/dcrdex/client/cmd/dexc/version.appPreRelease=%[1]s `+
-		`-X decred.org/dcrdex/client/cmd/dexcctl/version.appBuild=release `+
-		`-X decred.org/dcrdex/client/cmd/dexcctl/version.appPreRelease=%[1]s`,
```
Comment (chappjc) [inline main.go]: Dang I have this wrong in the code comments and my package script too.

**Comment by jrick** [comment]
```
2022/01/11 18:54:31 dist "dcrinstall-manifests" manifest hash: SHA256:eced88b76d63a4194adb58488badca379dbb8d1ae64fbe9893ff9acee7e8017f (go version go1.17.6 openbsd/amd64)
```

**Comment by jrick** [comment]
actually, this one

```
2022/01/11 20:01:42 dist "dcrinstall-manifests" manifest hash: SHA256:332050432abd26683080c30967244f0d7a1e856cb1cb9ffe01dca2e4b109dca6 (go version go1.17.6 openbsd/amd64)
```

**Comment by chappjc** [comment]
```
2022/01/11 18:10:53 dist: dist/dcrinstall-v1.7.0-rc2-manifests.txt
2022/01/11 18:10:53 dist "decred" manifest hash: SHA256:cb4dc9ed8f847f93ee82058cc4f25751cbb6e3ce484a30baffc37ca58a42c39d (go version go1.17.6 linux/amd64)
2022/01/11 18:10:53 dist "dexc" manifest hash: SHA256:c8e3e8df953cc344bfeab01451c9b42ead4319d5bfd06ed844c8cdbc2aa73940 (go version go1.17.6 linux/amd64)
2022/01/11 18:10:53 dist "dcrinstall" manifest hash: SHA256:b28b013293e4b16ba404a18ee5ab2c55e7b52d2f96c7c2db6de2de9aed30c7e4 (go version go1.17.6 linux/amd64)
2022/01/11 18:10:53 dist "dcrinstall-manifests" manifest hash: SHA256:332050432abd26683080c30967244f0d7a1e856cb1cb9ffe01dca2e4b109dca6 (go version go1.17.6 linux/amd64)
```

**Review by davecgh — APPROVED** [review]
```
windows: dist "dcrinstall-manifests" manifest hash: SHA256:332050432abd26683080c30967244f0d7a1e856cb1cb9ffe01dca2e4b109dca6 (go version go1.17.6 windows/amd64)
linux: dist "dcrinstall-manifests" manifest hash: SHA256:332050432abd26683080c30967244f0d7a1e856cb1cb9ffe01dca2e4b109dca6 (go version go1.17.6 linux/amd64)
```

**Pull request merged**